### PR TITLE
feat(linter): implement noOctalEscape

### PR DESCRIPTION
**Pull request by fireairforce**
## Summary

closes: https://github.com/biomejs/biome/issues/3929

### CODE REVIEW

File: crates/biome_js_analyze/src/lint/nursery/no_octal_escape.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
+use biome_analyze::{context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_js_syntax::JsStringLiteralExpression;
+use biome_rowan::AstNode;
+use regex::Regex;
+use std::sync::LazyLock;
+
+pub static OCTAL_ESCAPE_REGEX: LazyLock<Regex> = LazyLock::new(|| {
+    Regex::new(r#"^(?:[^\\]|\\.)*?\\([0-7]{1,3})"#).unwrap()
+});
```
Comment (Conaclos) [inline crates/biome_js_analyze/src/lint/nursery/no_octal_escape.rs:10]: Hi! Thanks for the quick PR!

Regexes have a large overhead. I was expecting an implementation similar to [no_useless_esapce_in_regex](https://github.com/biomejs/biome/blob/main/crates/biome_js_analyze/src/lint/nursery/no_useless_escape_in_regex.rs#L58).

Comment (fireairforce) [reply crates/biome_js_analyze/src/lint/nursery/no_octal_escape.rs:10]: ok, i will update this

Comment (fireairforce) [reply crates/biome_js_analyze/src/lint/nursery/no_octal_escape.rs]: updated~

File: crates/biome_js_analyze/src/lint/nursery/no_octal_escape.rs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                node.range(),
+                markup! {
+                    "Don't use "<Emphasis>"octal"</Emphasis>
+                },
+            )
+            .note(markup! {
+                "Don't use octal: " {text}
+            })
+            .note(markup! {
+                "Use '\\u...' instead."
```
Comment (dyc3) [inline crates/biome_js_analyze/src/lint/nursery/no_octal_escape.rs:80]: This message should elaborate a little bit more to explain that they should use unicode or hexidecimal escape sequences instead, or add an additional `\` if they want that character literally.

File: crates/biome_js_analyze/tests/specs/nursery/noOctalEscape/valid.js
Comment (dyc3) [inline crates/biome_js_analyze/tests/specs/nursery/noOctalEscape/valid.js:1]: `"\t\r\n"` should also be valid

**Review by dyc3 — CHANGES_REQUESTED** [review]

File: crates/biome_js_analyze/src/lint/nursery/no_octal_escape.rs (hunk 2 above)
Comment (fireairforce) [reply crates/biome_js_analyze/src/lint/nursery/no_octal_escape.rs]: add a (unicode) message here..

File: crates/biome_js_analyze/tests/specs/nursery/noOctalEscape/valid.js
Comment (fireairforce) [reply crates/biome_js_analyze/tests/specs/nursery/noOctalEscape/valid.js:1]: of course, i add this into test case.

File: crates/biome_js_analyze/src/lint/nursery/no_octal_escape.rs (hunk 2 above)
Comment (fireairforce) [reply crates/biome_js_analyze/src/lint/nursery/no_octal_escape.rs]: updated here...

**Review by dyc3 — APPROVED** [review]

**Review by Conaclos — APPROVED** [review]
We could improve the diagnostic by reporting only the octal escape sequence. This is a possible future improvement.

**Pull request merged**